flang/lib/Optimizer/Builder/IntrinsicCall.cpp

```diff
@@ -660,6 +664,9 @@ static constexpr IntrinsicHandler handlers[]{
      &I::genTranspose,
      {{{"matrix", asAddr}}},
      /*isElemental=*/false},
+    {"threadfence", &I::genThreadFence, {}, /*isElemental=*/false},
+    {"threadfence_block", &I::genThreadFenceBlock, {}, /*isElemental=*/false},
+    {"threadfence_system", &I::genThreadFenceSystem, {}, /*isElemental=*/false},
```

Comment:
This need to be just between "tan" and "trailz"

Reply:
Indeed. I am adding a static assert to help detecting that kind of issues in the future: https://github.com/llvm/llvm-project/pull/120399

Reply:
Thank you. Since most worked, I thought the order was just there for readability.

Reply:
Yeah it was not easy to spot since `threadfence_block` and the other worked. I guess it was just no luck that the pivot was probably around these.